inc/classes/subscriber/Addons/Cloudflare/CloudflareSubscriber.php

```diff
@@ -90,4 +123,412 @@ private function should_filter_varnish() {
+	}
+
+	/**
+	 * Purge Cloudflare cache URLs automatically if Cache Everything is set as a Page Rule
+	 *
+	 * @since 3.4.2
+	 * @author Soponar Cristina
+	 *
+	 * @param WP_Post $post       The post object.
+	 * @param array   $purge_urls URLs cache files to remove.
+	 * @param string  $lang       The post language.
+	 */
+	public function auto_purge_by_url( $post, $purge_urls, $lang ) {
+		if ( ! $this->options->get( 'do_cloudflare' ) || ! current_user_can( 'rocket_purge_cloudflare_cache' ) ) {
+			return;
+		}
+
+		$cf_cache_everything = $this->cloudflare->has_page_rule( 'cache_everything' );
+
+		if ( is_wp_error( $cf_cache_everything ) || ! $cf_cache_everything ) {
+			return;
+		}
+
+		// Purge CloudFlare.
+		$cf_purge = $this->cloudflare->purge_by_url( $post, $purge_urls, $lang );
+
+		if ( is_wp_error( $cf_purge ) ) {
+			$cf_purge_result = [
+				'result'  => 'error',
+				// translators: %s = CloudFare API return message.
+				'message' => sprintf( __( '<strong>WP Rocket:</strong> %s', 'rocket' ), $cf_purge->get_error_message() ),
+			];
+		} else {
+			$cf_purge_result = [
+				'result'  => 'success',
+				'message' => __( '<strong>WP Rocket:</strong> Cloudflare cache successfully purged.', 'rocket' ),
+			];
+		}
+
+		set_transient( get_current_user_id() . '_cloudflare_purge_result', $cf_purge_result );
+	}
+
+	/**
+	 * Purge CloudFlare cache
+	 *
+	 * @since 2.5
+	 */
+	public function purge_cache_no_die() {
+		if ( ! isset( $_GET['_wpnonce'] ) || ! wp_verify_nonce( sanitize_key( $_GET['_wpnonce'] ), 'rocket_purge_cloudflare' ) ) {
+			wp_nonce_ays( '' );
+		}
+		if ( ! $this->options->get( 'do_cloudflare' ) || ! current_user_can( 'rocket_purge_cloudflare_cache' ) ) {
+			return;
+		}
+
+		// Purge CloudFlare.
+		$cf_purge = $this->cloudflare->purge_cloudflare();
+
+		if ( is_wp_error( $cf_purge ) ) {
+			$cf_purge_result = [
+				'result'  => 'error',
+				// translators: %s = CloudFare API return message.
+				'message' => sprintf( __( '<strong>WP Rocket:</strong> %s', 'rocket' ), $cf_purge->get_error_message() ),
+			];
+		} else {
+			$cf_purge_result = [
+				'result'  => 'success',
+				'message' => __( '<strong>WP Rocket:</strong> Cloudflare cache successfully purged.', 'rocket' ),
+			];
+		}
+
+		set_transient( get_current_user_id() . '_cloudflare_purge_result', $cf_purge_result );
+	}
+
+	/**
+	 * Purge CloudFlare cache
+	 *
+	 * @since 2.5
+	 */
+	public function purge_cache() {
+		$this->purge_cache_no_die();
+		wp_safe_redirect( esc_url_raw( wp_get_referer() ) );
+		die();
+	}
+
+	/**
+	 * Set Real IP from CloudFlare
+	 *
+	 * @since 2.8.16 Uses CloudFlare API v4 to get CloudFlare IPs
+	 * @since 2.5.4
+	 * @source cloudflare.php - https://wordpress.org/plugins/cloudflare/
+	 */
+	public function set_real_ip() {
+		// only run this logic if the REMOTE_ADDR is populated, to avoid causing notices in CLI mode.
+		if ( ! isset( $_SERVER['HTTP_CF_CONNECTING_IP'] ) || ! isset( $_SERVER['REMOTE_ADDR'] ) ) {
```

Comment:
Optional, can be simplified:
```suggestion
		if ( ! isset( $_SERVER['HTTP_CF_CONNECTING_IP'], $_SERVER['REMOTE_ADDR'] ) ) {
```